text/0000-informational-commands.md

```diff
@@ -0,0 +1,302 @@
+- Feature Name: informational_commands
+- Start Date: 2019-05-15
+- RFC PR: (leave this empty)
+- Notion Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Add an informational command, `volta list`, replacing the `volta current` command and expanding on its capabilities. The `list` command will allow users to see their default and project-specific toolchains, and will support both pretty-printed user-facing output and at least one mode of tool-friendly output.
+
+# Motivation
+[motivation]: #motivation
+
+Why are we doing this? What use cases does it support? What is the expected outcome?
+
+# Pedagogy
+[pedagogy]: #pedagogy
+
+How should we explain and teach this feature to users? How should users understand this feature? This generally means:
+
+- Introducing new named concepts.
+- Explaining the feature largely in terms of examples.
+- Explaining how users should _think_ about the feature, and how it should impact the way they use Notion. It should explain the impact as concretely as possible.
+- If applicable, describe the differences between teaching this to existing Node programmers and new Node programmers.
+
+It is not necessary to write the actual feature documentation in this section, but it should establish the concepts on which the documentation would be built.
+
+# Details
+[details]: #details
+
+## User configuration
+
+Throughout, we will assume the user has the following configuration for illustrative purposes:
+
+- Node versions installed: v12.2.0, v11.9.0, v10.15.3 (default), v8.16.0
+- Yarn versions installed: v1.16.0, v1.12.3 (default)
+- Tools installed:
+    - ember-cli: v3.10.0 (default) on Node v12.2.0, v3.8.2 on Node v12.2.0, with binary `ember`
+    - typescript: v3.4.5 on Node v12.2.0, v3.0.3 (default) on Node v12.2.0, with binaries `tsc`, `tsserver`
+    - create-react-app: v3.0.1 (default) on Node v12.2.0, with binary `create-react-app`
+    - yarn-deduplicate: v1.1.1 on Node v12.2.0, with binary `yarn-deduplicate` (notice that this is not a *default*; assume the user ran `volta )
+
+They also have two projects with the following pins:
+
+- `~/pinned/node-only/package.json`: f
+
+    ```json
+    {
+      "volta": {
+        "node": "v8.16.0",
+      }
+    }
+    ```
+
+- `~/pinned/node-and-yarn/package.json`:
+
+    ```json
+    {
+      "volta": {
+        "node": "v12.2.0",
+        "yarn": "v1.16.0"
+      }
+    }
+    ```
+
+## Overview of CLI commands
+
+Top-level command: show the current to
+
+```sh
+$ volta list
+⚡️ Currently active toolchain:
+
+    Node: v8.16.0 (from `~/path/to/project/package.json`)
+    Yarn: v1.12.3 (default)
+    Package binaries available:
+        create-react-app, ember, tsc, tsserver, yarn-deduplicate
+
+See options for more detailed reports by running `volta list --help`.
+```
+
+Show all available tools, including binaries, with `list --all`:
+
+```sh
+$ volta list --all
+Node runtimes:
+    v12.2.0
+    v11.9.0
+    v10.15.3 (default)
+    v8.16.0
+
+Packagers:
```

Comment:
This list should probably include `npm` as well, since every node comes with one.

Reply:
That's an interesting point I've been mulling on as well. My thinking here was that we'd be listing *custom* packagers, rather than the defaults – which probably means listing it with the runtimes? But I'm very much up for changing my mind on how it should be handled.

Reply:
Added an unresolved question on this point.